pkg/metrics/filters/filters.go

```diff
@@ -0,0 +1,116 @@
+package filters
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apiserver/pkg/authentication/authenticatorfactory"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	"k8s.io/apiserver/pkg/authorization/authorizerfactory"
+	"k8s.io/apiserver/pkg/server/options"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+// WithAuthenticationAndAuthorization provides a metrics.Filter for authentication and authorization.
+// Metrics will be authenticated (via TokenReviews) and authorized (via SubjectAccessReviews) with the
+// kube-apiserver.
+// For the authentication and authorization the controller needs a ClusterRole
+// with the following rules:
+// - apiGroups:
+//   - authentication.k8s.io
+//     resources:
+//   - tokenreviews
+//     verbs:
+//   - create
+//
+// - apiGroups:
+//   - authorization.k8s.io
+//     resources:
+//   - subjectaccessreviews
+//     verbs:
+//   - create
+//
+// To scrape metrics e.g. via Prometheus the client needs a ClusterRole
+// with the following rule:
+// - nonResourceURLs:
+//   - "/metrics"
+//     verbs:
+//   - get
+//
+// Note: Please note that configuring this metrics provider will introduce a dependency to "k8s.io/apiserver"
+// to your go module.
+func WithAuthenticationAndAuthorization(config *rest.Config, httpClient *http.Client) (metrics.Filter, error) {
+	kubeClient, err := kubernetes.NewForConfigAndClient(config, httpClient)
```

Comment:
Lets use the specific clients we need to avoid building all clients for all apis when this is used:
```suggestion
	authnv1client, err := k8s.io/client-go/kubernetes/typed/authentication/v1.NewForConfigAndClient(config, httpClient)
	authzv1client, err := k8s.io/client-go/kubernetes/typed/authorization/v1.NewForConfigAndClient(config, httpClient
```

Reply:
Ah, thx! Didn't understood this the first time :)